packages/@aws-cdk/aws-lambda-python/lib/bundling.ts

```diff
@@ -16,137 +17,89 @@ export const BUNDLER_DEPENDENCIES_CACHE = '/var/dependencies';
-  // sources.
-  const dockerfile = hasDeps
-    ? 'Dockerfile.dependencies'
-    : 'Dockerfile';
-
-  // copy Dockerfile to workdir
-  fs.copyFileSync(path.join(__dirname, dockerfile), path.join(stagedir, dockerfile));
+export class Bundling implements CdkBundlingOptions {
+  public static bundle(options: BundlingProps): AssetCode {
+    return Code.fromAsset(options.entry, {
+      assetHash: options.assetHash,
+      assetHashType: options.assetHashType,
+      exclude: DEPENDENCY_EXCLUDES,
+      bundling: new Bundling(options),
+    });
+  }
 
-  const image = cdk.DockerImage.fromBuild(stagedir, {
-    buildArgs: {
-      IMAGE: runtime.bundlingImage.image,
-    },
-    platform: architecture.dockerPlatform,
-    file: dockerfile,
-  });
+  public readonly image: DockerImage;
+  public readonly command: string[];
 
-  return lambda.Code.fromAsset(entry, {
-    assetHashType: options.assetHashType,
-    assetHash: options.assetHash,
-    exclude: DEPENDENCY_EXCLUDES,
-    bundling: {
+  constructor(props: BundlingProps) {
+    const {
+      entry,
+      runtime = Runtime.PYTHON_3_7,
+      architecture = Architecture.X86_64, outputPathSuffix,
       image,
-      command: ['bash', '-c', depsCommand],
-    },
-  });
-}
-
-/**
- * Checks to see if the `entry` directory contains a type of dependency that
- * we know how to install.
- */
-export function stageDependencies(entry: string, stagedir: string): boolean {
-  const prefixes = [
-    'Pipfile',
-    'pyproject',
-    'poetry',
-    'requirements.txt',
-  ];
+    } = props;
+
+    const outputPath = path.join(AssetStaging.BUNDLING_OUTPUT_DIR, outputPathSuffix ?? '');
+
+    const bundlingCommands = this.createBundlingCommand({
+      entry,
+      inputDir: AssetStaging.BUNDLING_INPUT_DIR,
+      outputDir: outputPath,
+    });
+
+    const defaultImage = DockerImage.fromBuild(path.join(__dirname, '../lib'), {
+      buildArgs: {
+        IMAGE: runtime.bundlingImage.image,
+      },
+      platform: architecture.dockerPlatform,
+    });
+    this.image = image ?? defaultImage;
+    this.command = ['bash', '-c', chain(bundlingCommands)];
+  }
 
-  let found = false;
-  for (const file of fs.readdirSync(entry)) {
-    for (const prefix of prefixes) {
-      if (file.startsWith(prefix)) {
-        fs.copyFileSync(path.join(entry, file), path.join(stagedir, file));
-        found = true;
-      }
-    }
+  private createBundlingCommand(options: BundlingCommandOptions): string[] {
+    const packaging = Packaging.fromEntry(options.entry);
+    let bundlingCommands: string[] = [];
+    // bundlingCommands.push(packaging.installCommand ?? '');
+    bundlingCommands.push(packaging.exportCommand ?? '');
+    if (packaging.dependenciesFile) {
+      bundlingCommands.push(`python -m pip install -r ${DependenciesFile.PIP} -t ${options.outputDir}`);
+    };
+    bundlingCommands.push(`cp -R ${options.inputDir} ${options.outputDir}`);
+    return bundlingCommands;
```

Comment:
Determines what bundling commands we'll use based on the dependency files that are found in `entry`.